src/regex.cr

```diff
@@ -513,7 +513,8 @@ class Regex
       capture_number = (name_table[capture_offset].to_u16 << 8) | name_table[capture_offset + 1].to_u16
 
       name_offset = capture_offset + 2
-      name = String.new((name_table + name_offset).pointer(name_entry_size - 3))
+      checked = (name_table + name_offset)[0, (name_entry_size - 3)]
```

Comment:
```suggestion
      checked = name_table[name_offset, name_entry_size - 3]
```

Reply:
@straight-shoota  can you please explain to me the difference?
It seems that `name_table + name_offset` returns a new Slice, is this the reason for the changer?
I'm sorry, I'm asking a lot of questions because I'm curious to understand why.

Reply:
Sure. `slice + x` is essentially the same as `slice[x, slice.size - x]`. Both return a new slice. Instead of chaining two calls to advance the pointer and then limit the size, this can be done in a single call. The result is the same, but it avoids some unnecessary duplicate instructions.